processor/transformprocessor/README.md

```diff
@@ -0,0 +1,57 @@
+# Transform Processor
+
+Supported pipeline types: traces
+
+The transform processor modifies telemetry based on configuration using the Telemetry Query Language.
+It takes a list of queries which are performed in the order specified in the config.
+
+Queries are composed of the following parts
+- Path expressions: Fields within the incoming data can be referenced using expressions composed of the names as defined
+in the OTLP protobuf definition. e.g., `status.code`, `attributes["http.method"]`. If the path expression begins with
+`resource.` or `instrumentation_library.`, it will reference those values.
+  - The name `instrumentation_library` within OpenTelemetry is currently under discussion and may be changed in the future.
+- Literals: Strings, ints, and floats can be referenced as literal values
+- Function invocations: Functions can be invoked with arguments matching the function's expected arguments
+- Where clause: Telemetry to modify can be filtered by appending `where a <op> b`, with `a` and `b` being any of the above.
+
+Supported functions:
+- `set(target, value)` - `target` is a path expression to a telemetry field to set `value` into. `value` is any value type.
```

Comment:
Continuing discussion from 

https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/7047/commits/7745311f894161f23888fc306a05be730b92e867

Is nil an accepted literal value?

_Originally posted by @bogdandrutu in https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/6985#discussion_r777649332_

Reply:
Currently it isn't - it will need to be added to allow for arbitrary functions. Curious what do you think the behavior of a function like set(name, nil) should be - should it set the name to empty string? Currently the code will ignore the call due to use of val.(string) type assertion

Reply:
Answer:

I think we should fail during "validation" of the config instead of "no-op". And I think is a good long-term plan, to fail fast in cases like this, especially because when you deploy the collector you want telemetry to be processed, not like in the case of an API/SDK where the telemetry is more or less a commodity.